oap-server/server-cluster-plugin/cluster-etcd-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/etcd/ITClusterEtcdPluginTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.cluster.plugin.etcd;
+
+import com.google.gson.Gson;
+import java.net.URI;
+import java.util.List;
+import mousio.etcd4j.EtcdClient;
+import mousio.etcd4j.responses.EtcdKeysResponse;
+import org.apache.skywalking.oap.server.core.cluster.RemoteInstance;
+import org.apache.skywalking.oap.server.core.remote.client.Address;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Alan Lau
+ */
+public class ITClusterEtcdPluginTest {
+
+    private ClusterModuleEtcdConfig etcdConfig;
+
+    private EtcdClient client;
+
+    private EtcdCoordinator coordinator;
+
+    private Gson gson = new Gson();
+
+    private Address remoteAddress = new Address("10.0.0.1", 1000, false);
+    private Address selfRemoteAddress = new Address("10.0.0.2", 1001, true);
+
+    private Address internalAddress = new Address("10.0.0.3", 1002, false);
+
+    private static final String SERVICE_NAME = "my-service";
+
+    @Before
+    public void before() throws Exception {
+        etcdConfig = new ClusterModuleEtcdConfig();
+        etcdConfig.setServiceName(SERVICE_NAME);
+        client = new EtcdClient(URI.create("http://127.0.0.1:2379"));
```

Comment:
I think this could cause CI failure randomly because of port conflict. Please take a good example from here, @kezhenxu94 's PR of IT test. cc @JaredTan95 

https://github.com/apache/skywalking/blob/fac940cdbe08a90137e3e9899351846a8099219e/oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java#L70

Reply:
We are running CI in ASF shared Jenkins cluster. We can't guarantee the port is available. 

@kezhenxu94 We may need a discussion about that, the current pom, looks like a conflict risk too.

Reply:
You mean that i should change the port to a parameter which given from the start running to avoid port conflict.

Reply:
Apollo test failed.

Reply:
`Apollo` test failure partly because there is no official image, and not because of this. Port need to be assigned randomly. Because in ASF Jenkins, we are sharing OS with other tasks of other ASF projects.

Reply:
For `Apollo`, I have been working with @kezhenxu94 to check and fix it. But that is another thread somehow.

Reply:
ok

Reply:
@wu-sheng I have make the host & port randomly as what you suggest. Also all test cases passed.